MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,455 @@
+                into m
+                select (!noClassLead ? $@"
+    /// <summary><c>{m.Name}</c> extension.</summary>
+
+    [GeneratedCode(""{thisAssemblyName.Name}"", ""{thisAssemblyName.Version}"")]" : null) + $@"
+    public static partial class {m.Name}Extension
+    {{
+{string.Join(null, from mo in m.Overloads select mo.ToFullString())}
+    }}";
+
+            var template = $@"
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+//
+// Licensed under the Apache License, Version 2.0 (the ""License"");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an ""AS IS"" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+// This code was generated by a tool. Any changes made manually will be lost
+// the next time this code is regenerated.
+
+namespace MoreLinq.NoConflict
+{{
+{string.Join("\n", imports)}
+{string.Join("\n", classes)}
+}}
+            ";
+
+            Console.WriteLine(template.Trim()
+                                      // normalize line endings
+                                      .Replace("\r", string.Empty)
+                                      .Replace("\n", Environment.NewLine));
+        }
+
+        public static TypeKey CreateTypeKey(TypeSyntax root,
+                                              Func<string, TypeKey> abbreviator = null)
+        {
+            return Walk(root ?? throw new ArgumentNullException(nameof(root)));
+
+            TypeKey Walk(TypeSyntax ts) =>
+                ts is GenericNameSyntax gns
+                ? new GenericTypeKey(gns.Identifier.ToString(),
+                                        ImmutableList.CreateRange(gns.TypeArgumentList.Arguments.Select(Walk)))
+                : ts is IdentifierNameSyntax ins
+                ? abbreviator?.Invoke(ins.Identifier.ValueText) ?? new SimpleTypeKey(ins.ToString())
+                : ts is PredefinedTypeSyntax pts
+                ? new SimpleTypeKey(pts.ToString())
+                : ts is ArrayTypeSyntax ats
+                ? new ArrayTypeKey(Walk(ats.ElementType),
+                                    ImmutableList.CreateRange(from rs in ats.RankSpecifiers select rs.Rank))
+                : ts is NullableTypeSyntax nts
+                ? new NullableTypeKey(Walk(nts.ElementType))
+                : ts is TupleTypeSyntax tts
+                ? (TypeKey) new TupleTypeKey(ImmutableList.CreateRange(from te in tts.Elements select Walk(te.Type)))
+                : throw new NotSupportedException("Unhandled type: " + ts);
+        }
+
+        static T Read<T>(IEnumerator<T> e, Func<Exception> errorFactory = null)
```

Comment:
All users of `Read` pass the same `errorFactory`, so this parameter seems unnecessary.

Reply:
Except this method is totally generic and has no clue it's being used for a particular purpose such as reading arguments.